includes/class-wc-admin-reports-downloads-query.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Class for parameter-based downloads report querying.
+ *
+ * Example usage:
+ * $args = array(
+ *      'before'       => '2018-07-19 00:00:00',
+ *      'after'        => '2018-07-05 00:00:00',
+ *      'page'         => 2,
+ *      'products'     => array(1,2,3)
+ * );
+ * $report = new WC_Admin_Reports_Downloads_Query( $args );
+ * $mydata = $report->get_data();
+ *
+ * @package  WooCommerce Admin/Classes
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Reports_Downloads_Query
+ */
+class WC_Admin_Reports_Downloads_Query extends WC_Admin_Reports_Query {
+
+	/**
+	 * Valid fields for downloads report.
+	 *
+	 * @return array
+	 */
+	protected function get_default_query_vars() {
```

Comment:
Is this method needed outside of being used in the filter in `get_data()`?

Reply:
I double checked, and yes it is called elsewhere in the chain it looks like and I get some failing tests trying to remove it.